app/main.py

```diff
@@ -2,24 +2,67 @@
 
 
 class IntegerRange:
```

Comment:
The `IntegerRange` class should inherit from `ABC` to properly implement the descriptor protocol. Consider changing the class definition to `class IntegerRange(ABC):`.